float64 dropoffX
float64 dropoffY
float64 dropoffZ
float64 angleDiff
---